ghapi/actions.py

```diff
@@ -159,3 +160,21 @@ def set_git_user(api=None):
         email = 'github-actions[bot]@users.noreply.github.com'
     run(f'git config --global user.email "{email}"')
     run(f'git config --global user.name  "{user}"')
+
+# %% ../01_actions.ipynb 52
+class PagesNotEnabled(Exception): pass
+class PagesConfigError(Exception): pass
+
+def enable_pages_actions(api):
+    from urllib.error import HTTPError
+    msg="Please enable GitHub Pages to publish from the root of the `gh-pages` branch per these instructions - https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-from-a-branch"
+    try: res = api.enable_pages(branch='gh-pages')
+    except TokenNotAuthorized as e: raise e
+    except HTTPError: 
+        if api.repos.get().visibility == 'private': raise PagesNotEnabled(f'For security reasons, we cannot automatically enable GitHub Pages on private repos. {msg}')
+        else: raise PagesNotEnabled(f'Your repository does not have GitHub Pages enabled. {msg}') # enable_pages returns a string if pages exist.
+    else:
```

Comment:
With the `PagesNotEnabled` Error, I would just emit these both to Actions directly.  Since `enabe_pages` checks to see if GitHub Pages is enabled I am reasonably confident that the`PagesNotEnabled` exception is correct